text/0148-sampling-probability.md

```diff
@@ -0,0 +1,209 @@
+Specify a foundation for sampling techniques in OpenTelemetry.
+
+## Motivation
+
+In both tracing and metrics, there are widely known techniques for
+sampling events that, when performed correctly, enable ways to lower
+collection costs.  While sampling techniques vary, it is possible
+specify high-level interoperability requirements that producers and
+consumers of sampled data may follow to enable a wide range of
+sampling designs.
+
+## Explanation
+
+Consider a hypothetical telemetry signal in which an API event
+produces a unit of data that has one or more associated numbers.
+Using the OpenTelemetry Metrics data model terminolgy, we have two
+scenarios in which sampling is common.
+
+1. Counter events: Each event represents a count, signifying the change in a sum
+2. Histogram events: Each event represents an individual variable, signifying new membership in a distribution
+
+A Tracing Span event qualifies as both of these cases simultaneously.  It is
+a Counter event (of 1 span) and a Histogram event (of 1 latency measurement).
+
+In Metrics, [Statsd Counter and Histogram events meet this definition](https://github.com/statsd/statsd/blob/master/docs/metric_types.md#sampling).
+
+In both cases, the goal in sampling is to estimate something about the
+population of all events, using only the events that were chosen in
+the sample.  Sampling theory defines various kinds of "estimator",
+algorithms for calculating statistics about the population using just
+the sample data.  For the broad class of telemetry sampling
+application considered here, we need an estimator for the population
+total represented by each individual event.
+
+### Sample count adjustment
+
+The estimated population total divided by the individual event value
+equals the event's _adjusted count_.  The adjusted count is named
+`sample_count` where it appears as a field in a telemetry data type.
+The field indicates that the individual sample event is estimated to
+represent `sample_count` number of identical events in the population.
+
+A standard sampling adjustment will be defined, and for it to work
+there is one essential requirement.  The selection procedure must be
+_unbiased_, a statistical term meaning that the process is expected to
+give equal consideration to all possible outcomes.
+
+The specified sampling adjustment sets the `sample_count` of each
+sampled event to the inverse of the event's inclusion probability.
+Conveying the inverse of the inclusion probability is convenient for
+several reasons:
+
+- A `sample_count` of 1 indicates no sampling was performed
+- A `sample_count` of 0 indicates an unrepresentative event
+- Larger `sample_count` indicates greater representivity
+- Smaller `sample_count` indicates smaller representivity
+- The sum of `sample_count` in a sample equals the expected 
+  population size.
+  
+The zero `sample_count` value supports collecting values that were
+rejected from a sample to use as exemplars, which supports encoding of
```

Comment:
Does "rejected from a sample to use as exemplars" mean that the event is _not_ exemplary, or that it's an example of something being elided? The list above says that a `sample_count` of zero indicates the former, using different language.

Reply:
I don't see these two (not exemplary and elided) as any different. The point I was trying to establish is that when sampling is in effect, and when the `sample_count` field is being interpreted from a stream of events, the value of zero is meaningful and necessary to ensure that the sum of sample_count equals the expected population count if we are to record a span that was not sampled.

A span with a count of zero should be considered a span that happened once but is not counted. You might use that span to complete another trace (rooted somewhere else), or you might use that span as an "auxiliary variable" (a term from Survey Analysis).

The specific use-case I have in mind when writing this stems from the weighted sampling algorithms mentioned here and above. If we want a sampling algorithm to compute a fixed size sample, when the number of events that will happen during an interval is unknown, there will be a chance that some item is first selected for a sample and later rejected from a sample (see [here](https://dl.acm.org/doi/10.1145/3147.3165)).